pkg/component/etcd/monitoring_test.go

```diff
@@ -404,26 +404,26 @@ static_configs:
 
 	alertingRulesBackup = `  # etcd backup failure alerts
   - alert: KubeEtcdDeltaBackupFailed
-    expr: ((time() - etcdbr_snapshot_latest_timestamp{job="kube-etcd3-backup-restore-` + testRole + `",kind="Incr"} > bool 900) + (etcdbr_snapshot_required{job="kube-etcd3-backup-restore-` + testRole + `", kind="Incr"} >= bool 1) == 2) + on(pod,role) group_left 0 * (etcd_server_is_leader{job="kube-etcd3-` + testRole + `"} == 1 )
+    expr: sum(((time() - etcdbr_snapshot_latest_timestamp{job="kube-etcd3-backup-restore-` + testRole + `",kind="Incr"} > bool 900) * etcdbr_snapshot_required{job="kube-etcd3-backup-restore-` + testRole + `",kind="Incr"}) * on(pod, role) etcd_server_is_leader{job="kube-etcd3-` + testRole + `"}) > 0
     for: 15m
     labels:
       service: etcd
       severity: critical
       type: seed
       visibility: operator
     annotations:
-      description: No delta snapshot for the past at least 30 minutes taken by backup-restore leader.
+      description: No delta snapshot for at least last 15 minutes have been taken by backup-restore leader.
       summary: Etcd delta snapshot failure.
   - alert: KubeEtcdFullBackupFailed
-    expr: ((time() - etcdbr_snapshot_latest_timestamp{job="kube-etcd3-backup-restore-` + testRole + `",kind="Full"} > bool 86400) + (etcdbr_snapshot_required{job="kube-etcd3-backup-restore-` + testRole + `", kind="Full"} >= bool 1) == 2) + on(pod,role) group_left 0 * (etcd_server_is_leader{job="kube-etcd3-` + testRole + `"} == 1 )
+    expr: sum(((time() - etcdbr_snapshot_latest_timestamp{job="kube-etcd3-backup-restore-` + testRole + `",kind="Full"} > bool 86400) * etcdbr_snapshot_required{job="kube-etcd3-backup-restore-` + testRole + `",kind="Full"}) * on(pod, role) etcd_server_is_leader{job="kube-etcd3-` + testRole + `"}) > 0
```

Comment:
Generally, these queries are more advanced and they are hard to read for humans in this single line representation.

There is a recent feature of the Prometheus UI that it can format PromQL queries in some canonical way.
Could you try to format these queries and use a multiline string representation in the yaml structure so that they become more human readable?

Example for a canonically formatted multiline promql in yaml: https://github.com/gardener/gardener/blob/7ce9dbb276a14c2f874479e0004edf2c06c9dcba/pkg/component/monitoring/charts/bootstrap/prometheus-rules/metering.rules.stateful.yaml#L8-L14

Maybe create first a formatting commit, and then apply your changes, so that the diff of the second commit will be specific to your change (and will not mix in the formatting changes).

Furthermore, maybe not in the PR description, but in the commit message, could you describe the reasoning of the changes that you applied? It could help to double check during a review that the changes are really matching your intentions. You could also use screenshots (in the PR, or PR comment) to show the affect of the changes, if that is easier, but a concise description and maybe a short example is also feasible.

Now specifically to the diff:
- using a top level `sum` aggregation means that all the labels are aggregated away. Is this intended? The resulting alert will have fewer labels than it had before - so I think the role will be missing which was previously included e.g. in the job label. So, an alert notification email previously contained the job label, now I think it will not contain that.

Otherwise a commit message like: "Previously, the expression checked that the last snapshot is more than a day old, and a full backup is required, and the etcd instance is a leader. It checked this by using arithmetics and bool comparison expressions, and with this change it is performing an equivalent check by multiplying all the values and checking if the result is more than 0, i.e. none of the metrics was zero...." would facilitate the review.

Reply:
I addressed what you guided. Can you take a look in the commit message? Yes, `sum` aggregation was not needed. I removed them in the update.